### PR TITLE
Remove Managed Memory from PMF utility.

### PR DESCRIPTION
**Pull request by esclapez**

### CODE REVIEW

File: Utility/pmf.cpp [hunk 1]
```diff
@@ -87,25 +85,32 @@ PMF::read_pmf(const std::string& myfile)
...
+    sinput >> pmf_data.pmf_X[i];
+    for (int j = 0; j < pmf_data.pmf_M; j++) {
+      sinput >> pmf_data.pmf_Y[j * pmf_data.pmf_N + i];
     }
   }
 
-  PMF::d_pmf_X = PMF::pmf_X.dataPtr();
-  PMF::d_pmf_Y = PMF::pmf_Y.dataPtr();
+  pmf_data_g = (PmfData *) The_Device_Arena()->alloc(sizeof(pmf_data)); 
+#ifdef AMREX_USE_CUDA
```
Comment (drummerdoc) [inline Utility/pmf.cpp]: Do we need to add AMREX_USE_HIP conditions here as well?

**Review by drummerdoc — APPROVED** [review]

File: Utility/pmf.cpp (hunk 1 above)
Comment (esclapez) [reply Utility/pmf.cpp]: That is something I'm not entirely sure. Before we only had NVIDIA/CUDA device, now there are several options, but we still use the USE_CUDA = TRUE and AMREX_USE_CUDA everywhere. The hto2memcpy from AMReX is such that it works regardless, but maybe there is a more generic compiler `ifdef` we should use to see if we compile for GPU ?

Comment (drummerdoc) [reply Utility/pmf.cpp]: In the Tulip branch that has to be merged, I started the annoying task of doing 
#If defined(AMREX_USE_CUDA) || defined(AMREX_USE_HIP)
or
#If !defined(AMREX_USE_CUDA) && !defined(AMREX_USE_HIP)

We're going to have to propagate those generalize checks here too...

Comment (esclapez) [reply Utility/pmf.cpp]: Did you have to use a USE_HIP = TRUE in the makefile to have it compile with HIP ?

Comment (esclapez) [reply Utility/pmf.cpp]: Having a single compiler option to test if CUDA/HIP/DPC++ would be nice, rather than playing with logical combinations of three.

Comment (drummerdoc) [reply Utility/pmf.cpp]: Yes I did.  And yes, having something like AMREX_USE_GPU would be good. I'll suggest

Comment (jrood-nrel) [reply Utility/pmf.cpp]: This already exists.

Comment (drummerdoc) [reply Utility/pmf.cpp]: Hey, at least I guessed correctly :)

**Pull request merged**